classes/Earnings.php

```diff
@@ -90,11 +90,14 @@ public function prepare_order_earnings( int $order_id ) {
 		}
 
 		if ( is_array( $items ) && count( $items ) ) {
-			$per_earning_refund = $deducted_amount / count( $items );
+			
 			foreach ( $items as $item ) {
+		
 				$subtotal_price  = $item->regular_price;
 				$item_sold_price = $order_model->get_item_sold_price( $item->id, false );
 
+				$per_earning_refund = ( $deducted_amount * $subtotal_price ) / $order_details->total_price;
```

Comment:
Use try-catch to avoid fatal error divide by zero
```php
try {
 $per_earning_refund = ( $deducted_amount * $subtotal_price ) / $order_details->total_price;
} catch( $e ){
 $per_earning_refund = 0;
}
```